ci/configs/default.sh

```diff
@@ -1,3 +1,6 @@
+#!/usr/bin/env bash
```

Comment:
In commit "ci: export LC_ALL" (04eb50505be5700b374d552aaa814422b41ebc33)

I don't think the config fragments like `ci/configs/default.sh` should have hashbangs or export LC_ALL because they aren't marked executable and aren't designed to be run as standalone scripts. (They only set variables and I wouldn't want to encourage them to do anything more than that.)

To avoid the lint error, I'd suggest renaming the config files to `ci/configs/*.bash` instead. This would also be nice because it would indicate that the config files must use bash, not some other variant of shell script. (Note that renaming would also require updating `.github/workflows/ci.yml` and `ci/README.md`)